happi/cli.py

```diff
@@ -0,0 +1,110 @@
+"""
+Module for a command line interface to happi
+"""
+# cli.py
+# NOTE: Does not work from any environment, must have pcds_conda activated
+## Can this script activate/deactivate pcds_conda upon use?
+
+import argparse
+import sys
+import happi.backends.json_db as json_db
+import happi.client as hcl
+
+parser = argparse.ArgumentParser(description='happi command line tool')
+
+# First add search argument
+parser.add_argument('--search', action='store_true',
+                    help='search for device')
+parser.add_argument('--field', help='field in happi database to consider')
+parser.add_argument('--value', help="value of 'field' to return results",
+                    type=str)
+
+parser.add_argument('--add', action='store_true',
+                    help='add a device')
+
+# TODO: Implement edit option
+#parser.add_argument('--edit', action='store_true',
+#                    help='edit a device')
+
+#args = parser.parse_args()
+#arg_dict = vars(args)
+
+def happi_cli(args):
+    args = parser.parse_args(args)
+    print(args)
+    arg_dict = vars(args)
+    print(arg_dict)
+    # Instantiate Client and JSONBackend
+    json_backend = json_db.JSONBackend('/home/sheppard/PCDSGitHub/device_config/db.json')
```

Comment:
We don't want this path hard-coded. I would use `happi.Client.from_config`. Checkout the tests to see how to set it up yourself 😉

Reply:
I'm not sure if I did this quite right. After looking at `client.from_config` and the implementation used in the tests I thought I should create a file `happi/happi.cfg` (I believe this gets created and removed during your tests), but this contains a line pointing to my hard-coded `db.json` file. For actual deployment I figured it could point to the real location of `device_config/db.json` but that might not have been what you had in mind.